src/mame/drivers/m72.cpp

```diff
@@ -1898,11 +1899,7 @@ MACHINE_CONFIG_START(m72_state::m72_xmultipl)
 	m72_8751(config);
 	MCFG_DEVICE_MODIFY("maincpu")
 	MCFG_DEVICE_PROGRAM_MAP(xmultiplm72_map)
-
-	MCFG_DEVICE_MODIFY("soundcpu")
-	MCFG_DEVICE_PERIODIC_INT_DRIVER(m72_state, nmi_line_pulse, 128*55) /* clocked by V1? (Vigilante) */
-								/* IRQs are generated by main Z80 and YM2151 */
-
+	/* Sound NMI is already generated by MCU */
```

Comment:
As noted in ccb1f70feaffcf4edc40ae7c4e4c71627ee43eda, I don't think the M72 driver's Z80 NMI handling is technically correct. P1 is just a quasi-bidirectional parallel port (with no built-in strobes) on the 8751, so the hardware can't possibly be producing the NMI the way MAME does now. What I think would be enough to correct this, though it would entail considerable refactoring, is modifying the data flow of sound commands and sample data to go from V30 to Z80 to 8751 to DAC instead of from V30 to 8751 to Z80 to DAC, since the Z80 NMI routine mostly just passes data through and the DAC doesn't require a write strobe.